content/blog/2023/2023-03-31-serialize-deserialize-with-jsonview.md

```diff
@@ -0,0 +1,617 @@
+
+    }
+    // Allowed fields for PATCH requests
+    public static interface PatchView {
+
+    }
+}
+````
+In the further sections, we will take a look at how to use `@JsonView` to cater to each of the usecases we previously looked at.
+
+{{% info title="Difference between @JsonView and @JsonIgnore" %}}
+- `@JsonIgnoreProperties` used at the class level is used to ignore multiple fields during both serialization and deserialization process.
+- `@JsonIgnore` can be used at getter or setter for a property to ignore the fields during deserialization or serialization respectively.
+- `@JsonView` is an enhancement over `@JsonIgnore` since we can selectively decide if a field needs to be ignored or not for a particular API.
+{{% /info %}}
+
+Let us define the model `UserData` class.
+````java
+public class UserData {
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.UserSummary.class, Views.ExternalView.class})
+    private long id;
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.UserSummary.class, Views.ExternalView.class})
+    private String firstName;
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.UserSummary.class, Views.ExternalView.class})
+    private String lastName;
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.UserSummary.class, Views.ExternalView.class})
+    private String dob;
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.PatchView.class, 
+                    Views.UserSummary.class, Views.ExternalView.class})
+    private String address;
+
+    @JsonView(value = 
+            {Views.GetView.class, Views.PatchView.class, 
+                    Views.UserSummary.class, Views.ExternalView.class})
+    private String suburb;
+
+    @JsonView(value = {Views.GetView.class, Views.PatchView.class, 
+            Views.UserSummary.class, Views.ExternalView.class})
+    private String city;
+
+    private boolean isInternalUser;
+
+    private String additionalData;
+
+    @JsonView(value = {Views.GetView.class, Views.InternalView.class})
+    private String loginId;
+
+    @JsonView(value = {Views.GetView.class, Views.InternalView.class})
+    private String loginPassword;
+    @JsonView(value = {Views.GetView.class, Views.InternalView.class})
+    private String ssnNumber;
+
+    @JsonView(Views.UserDetailedSummary.class)
+    private String createdBy;
+
+    @JsonView(Views.UserDetailedSummary.class)
+    private LocalDate createdDate;
+
+    @JsonView(Views.UserDetailedSummary.class)
+    private String updatedBy;
+
+    @JsonView(Views.UserDetailedSummary.class)
+    private LocalDate updatedDate;
+    // Code for getters and setters
+}
+````
+As seen above, depending on our use case we have defined fields to have multiple views. 
+Next, let us take a look at how the controller class will use them.
+
+````java
+@RestController
+@RequestMapping("/internal")
+public class InternalUserController {
+
+    private static final Logger log = 
```

Comment:
The section regarding logging in and service injection can be omitted as the complete code can be found in the code examples on our GitHub repository.